data/transactions/logic/assembler_test.go

```diff
@@ -518,20 +518,47 @@ func testMatch(t testing.TB, actual, expected string) bool {
 	}
 }
 
-func assemblyTrace(text string, ver uint64) string {
+func assembleWithTrace(text string, ver uint64) (*OpStream, error) {
 	ops := newOpStream(ver)
 	ops.Trace = &strings.Builder{}
-	ops.assemble(text)
-	return ops.Trace.String()
+	err := ops.assemble(text)
+	return &ops, err
+}
+
+func lines(s string, num int) (bool, string) {
+	if num < 1 {
+		return true, ""
+	}
+	found := 0
+	for i := 0; i < len(s); i++ {
```

Comment:
Potentially more Goish to range over s
```suggestion
	for i, r := range  s{
```

Reply:
When you range of a `string` Go does unicode interpretation. I don't think that will muck anything up, but I don't want it.